### ISSUE TITLE
Mesosphere DCOS - DCOS cli HDFS deployment gets stuck

### ISSUE THREAD

**Issue by akamalov** [mesosphere/hdfs-deprecated#262]
Greetings,

Mesosphere DCOS v1.7, attempting to deploy HDFS using dcos cli. I am getting indefinite "Deploying" status where HDFS deploys couple of journal nodes, data nodes and then gets stuck in "Deploying" state. Has anyone  had issues similar to this ? Tried to follow log entries with `dcos service log --follow hdfs`, getting nothing special which would point me to an error or something. Tailing the log giving me the following:

```
INFO  [2016-07-01 10:35:48,874] com.mesosphere.dcos.hdfs.scheduler.HdfsRepairScheduler: No Tasks found requiring replacement
INFO  [2016-07-01 10:35:48,874] com.mesosphere.dcos.hdfs.scheduler.HdfsScheduler: Stage scheduler accepted offers: ids = []
INFO  [2016-07-01 10:35:48,874] com.mesosphere.dcos.hdfs.scheduler.HdfsScheduler: Declining offers: ids = [6a10bd5d-1629-43fe-aea8-2374c40ec3b1-O1442957]
INFO  [2016-07-01 10:35:48,874] com.mesosphere.dcos.hdfs.scheduler.tasks.HdfsTasks: Executing condemned tasks
INFO  [2016-07-01 10:35:48,874] com.mesosphere.dcos.hdfs.scheduler.tasks.HdfsTasks: No condemned tasks to execute
INFO  [2016-07-01 10:35:49,791] com.mesosphere.dcos.hdfs.scheduler.HdfsScheduler: Received 1 offers
INFO  [2016-07-01 10:35:49,791] com.mesosphere.dcos.hdfs.scheduler.HdfsScheduler: Executing Block: id = c30188cb-b93b-46c1-99ab-38ead76e7b44, name =journalnode-0, offers = [6a10bd5d-1629-43fe-aea8-2374c40ec3b1-O1442959]
INFO  [2016-07-01 10:35:49,791] org.apache.mesos.scheduler.plan.DefaultStageScheduler: Processing resource offers for block: journalnode-0
INFO  [2016-07-01 10:35:49,792] com.mesosphere.dcos.hdfs.scheduler.plan.HdfsTaskBlock: Block creating new task: name = journalnode-0, id = c30188cb-b93b-46c1-99ab-38ead76e7b44, task = JournalNodeTask{info = name: "journalnode-0"
task_id {
  value: "journalnode-0_00df2de7-e261-4b3f-8d5f-7f3f53dfcfd9"
}
slave_id {
  value: ""
}
resources {
  name: "cpus"
  type: SCALAR
  scalar {
    value: 0.5
  }
  role: "hdfs"
  reservation {
    principal: "hdfs"
  }
}
resources {
  name: "mem"
  type: SCALAR
  scalar {
    value: 2048.0
  }
  role: "hdfs"
  reservation {
    principal: "hdfs"
  }
}
resources {
  name: "disk"
  type: SCALAR
  scalar {
    value: 10240.0
  }
  role: "hdfs"
  disk {
    persistence {
      id: ""
      principal: "hdfs"
    }
    volume {
      container_path: "volume"
      mode: RW
    }
  }
  reservation {
    principal: "hdfs"
  }
}
resources {
  name: "ports"
  type: RANGES
  ranges {
    range {
      begin: 8480
      end: 8480
    }
    range {
      begin: 8485
      end: 8485
    }
  }
  role: "hdfs"
  reservation {
    principal: "hdfs"
  }
}
executor {
  executor_id {
    value: "hdfs-executorjournalnode-0-247bca8b-a9d7-4ce7-88e3-807330a157c2"
  }
  resources {
    name: "cpus"
    type: SCALAR
    scalar {
      value: 0.5
    }
    role: "hdfs"
    reservation {
      principal: "hdfs"
    }
  }
  resources {
    name: "mem"
    type: SCALAR
    scalar {
      value: 1024.0
    }
    role: "hdfs"
    reservation {
      principal: "hdfs"
    }
  }
  resources {
    name: "disk"
    type: SCALAR
    scalar {
      value: 1024.0
    }
    role: "hdfs"
    reservation {
      principal: "hdfs"
    }
  }
  command {
    uris {
      value: "https://downloads.mesosphere.com/hdfs/assets/0.9.0-2.6.0/hadoop-2.6.0-cdh5.7.1-dcos.tar.gz"
      executable: false
      extract: true
      cache: false
    }
    uris {
      value: "https://downloads.mesosphere.com/hdfs/assets/0.9.0-2.6.0/executor.zip"
      executable: false
      extract: true
      cache: false
    }
    uris {
      value: "https://downloads.mesosphere.com/hdfs/assets/0.9.0-2.6.0/jre-8u91-linux-x64.tar.gz"
      executable: false
      extract: true
      cache: false
    }
    environment {
      variables {
        name: "HADOOP_DEFAULT_NAME"
        value: "hdfs://hdfs"
      }
      variables {
        name: "HADOOP_HUE_HOSTS"
        value: "*"
      }
      variables {
        name: "HADOOP_HUE_GROUPS"
        value: "*"
      }
      variables {
        name: "HADOOP_ROOT_HOSTS"
        value: "*"
      }
      variables {
        name: "HADOOP_ROOT_GROUPS"
        value: "*"
      }
      variables {
        name: "HADOOP_HTTP_FS_GROUPS"
        value: "*"
      }
      variables {
        name: "HADOOP_HTTP_FS_HOSTS"
        value: "*"
      }
      variables {
        name: "SERVICE_NAME"
        value: "hdfs"
      }
      variables {
        name: "HDFS_NAME_NODE_BIND_HOST"
        value: "0.0.0.0"
      }
      variables {
        name: "HDFS_NAME_NODE_RPC_PORT"
        value: "9001"
      }
      variables {
        name: "HDFS_NAME_NODE_HTTP_PORT"
        value: "9002"
      }
      variables {
        name: "HDFS_JOURNAL_NODES"
        value: "3"
      }
      variables {
        name: "HDFS_JOURNAL_NODE_ADDRESS"
        value: "0.0.0.0"
      }
      variables {
        name: "HDFS_JOURNAL_NODE_RPC_PORT"
        value: "8485"
      }
      variables {
        name: "HDFS_JOURNAL_NODE_HTTP_PORT"
        value: "8480"
      }
      variables {
        name: "HDFS_DATA_NODE_ADDRESS"
        value: "0.0.0.0"
      }
      variables {
        name: "HDFS_DATA_NODE_RPC_PORT"
        value: "9003"
      }
      variables {
        name: "HDFS_DATA_NODE_IPC_PORT"
        value: "9005"
      }
      variables {
        name: "HDFS_DATA_NODE_HTTP_PORT"
        value: "9004"
      }
      variables {
        name: "HDFS_VOLUME_DIRECTORY"
        value: "volume"
      }
      variables {
        name: "HDFS_DOMAIN_SOCKET_DIRECTORY"
        value: ""
      }
      variables {
        name: "HDFS_ZOOKEEPER_QUORUM"
        value: "master.mesos:2181"
      }
      variables {
        name: "HDFS_PERMISSIONS_ENABLED"
        value: "false"
      }
      variables {
        name: "HDFS_DATA_NODE_BANDWIDTH_PER_SECOND"
        value: "41943040"
      }
      variables {
        name: "HDFS_NAME_NODE_THRESHOLD_PERCENTAGE"
        value: "0.9"
      }
      variables {
        name: "HDFS_NAME_NODE_HEARTBEAT_RECHECK_INTERVAL"
        value: "60000"
      }
      variables {
        name: "HDFS_DATA_NODE_HANDLER_COUNT"
        value: "10"
      }
      variables {
        name: "HDFS_NAME_NODE_HANDLER_COUNT"
        value: "20"
      }
      variables {
        name: "HDFS_COMPRESS_IMAGE"
        value: "true"
      }
      variables {
        name: "HDFS_IMAGE_COMPRESSION_CODEC"
        value: "org.apache.hadoop.io.compress.SnappyCodec"
      }
      variables {
        name: "HDFS_NAME_NODE_INVALIDATE_WORK_PERCENTAGE"
        value: "0.95"
      }
      variables {
        name: "HDFS_NAME_NODE_REPLICATION_WORK_MULTIPLIER"
        value: "4"
      }
      variables {
        name: "HDFS_CLIENT_READ_SHORT_CIRCUIT"
        value: "true"
      }
      variables {
        name: "HDFS_CLIENT_READ_SHORT_CIRCUIT_STREAMS"
        value: "1000"
      }
      variables {
        name: "HDFS_CLIENT_READ_SHORT_CIRCUIT_CACHE_EXPIRY_MS"
        value: "1000"
      }
      variables {
        name: "JAVA_HOME"
        value: "./jre1.8.0_91"
      }
      variables {
        name: "JAVA_OPTS"
        value: "-Xmx768M"
      }
      variables {
        name: "HDFS_HOME_DIR"
        value: "./hadoop-2.6.0-cdh5.7.1"
      }
      variables {
        name: "HDFS_VERSION"
        value: "2.5.0"
      }
    }
    value: "./executor/bin/hdfs-executor executor/conf/executor.yml"
  }
  framework_id {
    value: "6a10bd5d-1629-43fe-aea8-2374c40ec3b1-0008"
  }
  name: "hdfs-executorjournalnode-0"
}
labels {
  labels {
    key: "hdfs_zkfc_formatted"
    value: "true"
  }
  labels {
    key: "hdfs_node_initialized"
    value: "true"
  }
  labels {
    key: "hdfs_firstlaunch"
    value: "true"
  }
  labels {
    key: "hdfs_task_type"
    value: "JOURNAL_NODE"
  }
  labels {
    key: "hdfs_task_hostname"
    value: ""
  }
  labels {
    key: "hdfs_task_state"
    value: "TASK_STAGING"
  }
  labels {
    key: "hdfs_fs_formatted"
    value: "true"
  }
  labels {
    key: "hdfs_heap_mb"
    value: "2048"
  }
  labels {
    key: "hdfs_upgrade"
    value: "false"
  }
  labels {
    key: "hdfs_standby"
    value: "false"
  }
  labels {
    key: "hdfs_task_ports"
    value: "8480,8485"
  }
  labels {
    key: "hdfs_rollback"
    value: "false"
  }
  labels {
    key: "hdfs_disk_type"
    value: "ROOT"
  }
}
}
INFO  [2016-07-01 10:35:49,793] org.apache.mesos.offer.MesosResourcePool: Retrieving resource for reservation
INFO  [2016-07-01 10:35:49,793] org.apache.mesos.offer.OfferEvaluator: Satisfying resource requirement: name: "cpus"
type: SCALAR
scalar {
  value: 0.5
}
role: "hdfs"
reservation {
  principal: "hdfs"
}
with resource: name: "cpus"
type: SCALAR
scalar {
  value: 0.5
}
role: "*"

INFO  [2016-07-01 10:35:49,793] org.apache.mesos.offer.OfferEvaluator: Reserves Resource
INFO  [2016-07-01 10:35:49,793] org.apache.mesos.offer.OfferEvaluator: Fulfilled resource: name: "cpus"
type: SCALAR
scalar {
  value: 0.5
}
role: "hdfs"
reservation {
  principal: "hdfs"
  labels {
    labels {
      key: "resource_id"
      value: "afb83c0e-ceca-4d3e-b34b-7f13693c9952"
    }
  }
}

INFO  [2016-07-01 10:35:49,793] org.apache.mesos.offer.MesosResourcePool: Retrieving resource for reservation
INFO  [2016-07-01 10:35:49,793] org.apache.mesos.offer.OfferEvaluator: Satisfying resource requirement: name: "mem"
type: SCALAR
scalar {
  value: 1024.0
}
role: "hdfs"
reservation {
  principal: "hdfs"
}
with resource: name: "mem"
type: SCALAR
scalar {
  value: 1024.0
}
role: "*"

INFO  [2016-07-01 10:35:49,794] org.apache.mesos.offer.OfferEvaluator: Reserves Resource
INFO  [2016-07-01 10:35:49,794] org.apache.mesos.offer.OfferEvaluator: Fulfilled resource: name: "mem"
type: SCALAR
scalar {
  value: 1024.0
}
role: "hdfs"
reservation {
  principal: "hdfs"
  labels {
    labels {
      key: "resource_id"
      value: "bf1fc1ec-9b6a-43b7-adbc-6fa889f1ab17"
    }
  }
}

INFO  [2016-07-01 10:35:49,794] org.apache.mesos.offer.MesosResourcePool: Retrieving resource for reservation
INFO  [2016-07-01 10:35:49,794] org.apache.mesos.offer.OfferEvaluator: Satisfying resource requirement: name: "disk"
type: SCALAR
scalar {
  value: 1024.0
}
role: "hdfs"
reservation {
  principal: "hdfs"
}
with resource: name: "disk"
type: SCALAR
scalar {
  value: 1024.0
}
role: "*"

INFO  [2016-07-01 10:35:49,794] org.apache.mesos.offer.OfferEvaluator: Reserves Resource
INFO  [2016-07-01 10:35:49,794] org.apache.mesos.offer.OfferEvaluator: Fulfilled resource: name: "disk"
type: SCALAR
scalar {
  value: 1024.0
}
role: "hdfs"
reservation {
  principal: "hdfs"
  labels {
    labels {
      key: "resource_id"
      value: "ffbb15be-a91b-479f-b6b9-67016a1207a9"
    }
  }
}

INFO  [2016-07-01 10:35:49,794] org.apache.mesos.offer.MesosResourcePool: Retrieving resource for reservation
INFO  [2016-07-01 10:35:49,794] org.apache.mesos.offer.OfferEvaluator: Satisfying resource requirement: name: "cpus"
type: SCALAR
scalar {
  value: 0.5
}
role: "hdfs"
reservation {
  principal: "hdfs"
}
with resource: name: "cpus"
type: SCALAR
scalar {
  value: 0.5
}
role: "*"

INFO  [2016-07-01 10:35:49,794] org.apache.mesos.offer.OfferEvaluator: Reserves Resource
INFO  [2016-07-01 10:35:49,794] org.apache.mesos.offer.OfferEvaluator: Fulfilled resource: name: "cpus"
type: SCALAR
scalar {
  value: 0.5
}
role: "hdfs"
reservation {
  principal: "hdfs"
  labels {
    labels {
      key: "resource_id"
      value: "8e0c007c-f7e9-45f4-adb9-99c05a49663d"
    }
  }
}

INFO  [2016-07-01 10:35:49,794] org.apache.mesos.offer.MesosResourcePool: Retrieving resource for reservation
INFO  [2016-07-01 10:35:49,794] org.apache.mesos.offer.OfferEvaluator: Satisfying resource requirement: name: "mem"
type: SCALAR
scalar {
  value: 2048.0
}
role: "hdfs"
reservation {
  principal: "hdfs"
}
with resource: name: "mem"
type: SCALAR
scalar {
  value: 2048.0
}
role: "*"

INFO  [2016-07-01 10:35:49,795] org.apache.mesos.offer.OfferEvaluator: Reserves Resource
INFO  [2016-07-01 10:35:49,795] org.apache.mesos.offer.OfferEvaluator: Fulfilled resource: name: "mem"
type: SCALAR
scalar {
  value: 2048.0
}
role: "hdfs"
reservation {
  principal: "hdfs"
  labels {
    labels {
      key: "resource_id"
      value: "f6a6def8-7d9f-40d0-b5c2-f383bf898027"
    }
  }
}

INFO  [2016-07-01 10:35:49,795] org.apache.mesos.offer.MesosResourcePool: Retrieving resource for reservation
WARN  [2016-07-01 10:35:49,795] org.apache.mesos.offer.OfferEvaluator: Failed to satisfy resource requirement: name: "disk"
type: SCALAR
scalar {
  value: 10240.0
}
role: "hdfs"
disk {
  persistence {
    id: ""
    principal: "hdfs"
  }
  volume {
    container_path: "volume"
    mode: RW
  }
}
reservation {
  principal: "hdfs"
}

WARN  [2016-07-01 10:35:49,795] org.apache.mesos.offer.OfferAccepter: No recommendations, nothing to do
INFO  [2016-07-01 10:35:49,795] com.mesosphere.dcos.hdfs.scheduler.plan.HdfsTaskBlock: Block c30188cb-b93b-46c1-99ab-38ead76e7b44 set status: status = Pending
INFO  [2016-07-01 10:35:49,795] com.mesosphere.dcos.hdfs.scheduler.HdfsScheduler: Stage scheduler accepted offers: ids = []
INFO  [2016-07-01 10:35:49,795] com.mesosphere.dcos.hdfs.scheduler.HdfsScheduler: Executing repair: offers = [6a10bd5d-1629-43fe-aea8-2374c40ec3b1-O1442959]
INFO  [2016-07-01 10:35:49,795] com.mesosphere.dcos.hdfs.scheduler.HdfsRepairScheduler: Performing repair: ignore = [journalnode-0]
INFO  [2016-07-01 10:35:49,795] com.mesosphere.dcos.hdfs.scheduler.HdfsRepairScheduler: Received offer: offer = id {
  value: "6a10bd5d-1629-43fe-aea8-2374c40ec3b1-O1442959"
}
framework_id {
  value: "6a10bd5d-1629-43fe-aea8-2374c40ec3b1-0008"
}
slave_id {
  value: "e05fba38-b8b5-4d00-b9de-abd48130b7d1-S1"
}
hostname: "192.168.120.162"
resources {
  name: "ports"
  type: RANGES
  ranges {
    range {
      begin: 1025
      end: 2180
    }
    range {
      begin: 2182
      end: 3887
    }
    range {
      begin: 3889
      end: 4139
    }
    range {
      begin: 4141
      end: 5049
    }
    range {
      begin: 5052
      end: 8079
    }
    range {
      begin: 8082
      end: 8180
    }
    range {
      begin: 8182
      end: 8999
    }
    range {
      begin: 9001
      end: 9091
    }
    range {
      begin: 9093
      end: 9989
    }
    range {
      begin: 9991
      end: 11311
    }
    range {
      begin: 11313
      end: 31727
    }
    range {
      begin: 31729
      end: 32000
    }
  }
  role: "*"
}
resources {
  name: "disk"
  type: SCALAR
  scalar {
    value: 1276.0
  }
  role: "*"
}
resources {
  name: "cpus"
  type: SCALAR
  scalar {
    value: 2.75
  }
  role: "*"
}
resources {
  name: "mem"
  type: SCALAR
  scalar {
    value: 13553.0
  }
  role: "*"
}
url {
  scheme: "http"
  address {
    hostname: "192.168.120.162"
    ip: "192.168.120.162"
    port: 5051
  }
  path: "/slave(1)"
}

INFO  [2016-07-01 10:35:49,795] com.mesosphere.dcos.hdfs.scheduler.HdfsRepairScheduler: No Tasks found requiring replacement
INFO  [2016-07-01 10:35:49,796] com.mesosphere.dcos.hdfs.scheduler.HdfsScheduler: Stage scheduler accepted offers: ids = []
INFO  [2016-07-01 10:35:49,796] com.mesosphere.dcos.hdfs.scheduler.HdfsScheduler: Declining offers: ids = [6a10bd5d-1629-43fe-aea8-2374c40ec3b1-O1442959]
INFO  [2016-07-01 10:35:49,796] com.mesosphere.dcos.hdfs.scheduler.tasks.HdfsTasks: Executing condemned tasks
INFO  [2016-07-01 10:35:49,796] com.mesosphere.dcos.hdfs.scheduler.tasks.HdfsTasks: No condemned tasks to execute
...
```

Any pointers on how one should tackle this ?

Thanks!!

**Comment by tapanhalani** [comment]
going through the same problem..

**Comment by todormazgalov** [comment]
@akamalov : There's a line in the log file "Failed to satisfy resource requirement: name: "disk""
Where exactly does the deployment get stuck? Do you have all the data nodes or its stuck when you try to create one of them? By default it tries to start 3 data nodes with 10 GB disk space. You can check if you have the space.

I have a similar problem. But my offers are being rejected with the following messages : 
Failed to satisfy resource requirement: name: "mem"
Failed to satisfy resource requirement: name: "cpus"
Failed to satisfy resource requirement: name: "ports"
even though I'm sure that I have the resources needed for the hdfs nodes.

My theory is: Since I was able to install HDFS few weeks ago with the same resources its not because of them. I've uninstalled it and after a while I tried installing it again. It managed to start few of the Journal Nodes( the deploy starts with them) but it stuck at the name node. The reason was that the ports were "taken" even though I've checked that they are available.

That's why I had to uninstall hdfs during the deploy process :(
The second try was also unsuccessful because the ports of the journal nodes were "taken"( the same way as the name node).
The third time I tried switching up the ports in the Advanced Installation form but it was stuck again. This time was because of the cpus and the memory.

Bottom line is: I think that the reason for this is Mesos and its offers. Somehow it doesn't understand that the deployment has stopped and that the resources are not taken anymore. Mesos doesn't give the right offers with all of the available resources.
I've tried cleaning up zookeeper, restarting some of the nodes, but nothing seems to be working. Using snapshots is always a solution, but I don't want to lose the current state of the cluster.

Does anyone have and idea how to fix this?

**Comment by tapanhalani** [comment]
The rejection of offers due to inability to satisfy resources is definitely a problem. But i am having a different issue, since all my resource offers are accepted. I have seen stdout and stderr, in which it becomes clear that after starting the hdfs swevice at port 9000 of the desired node, some error occurs with the GET requests to org.apache.mesos.scheduler, causing 503 response and hence a 500 internal server error. And everytime it is happening the same way and gwtting the same error.

**Comment by triclambert** [comment]
Please note that this repository is now deprecated.  See [this issue](https://github.com/mesosphere/hdfs/issues/270#issuecomment-247233350) for a more detailed explanation.
